cdap-ui/app/cdap/components/KeyValuePairs/KeyValueStore.js

```diff
@@ -32,13 +32,15 @@ const keyValues = (state = [], action=defaultAction) => {
         return stateCopy;
       }
       stateCopy.pairs[action.payload.index].key = action.payload.key;
+      stateCopy.pairs[action.payload.index].valueChanged = true;
```

Comment:
We don't right now. Should we? I implemented it this way because we don't want to do a network request to compare the new values with the resolved values every time the user types, just to determine whether or not to show Reset. But maybe we can debounce it...